parser/strconv.go

```diff
@@ -0,0 +1,22 @@
+package parser
+
+import (
+	"time"
+
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/internal/parser"
+)
+
+func ParseTime(lit string) (time.Time, error) {
```

Comment:
I don't like the stutter of `parser.ParseTime` from an API usage point of view.

Reply:
Should we rename all the parse methods in the parser package to not include the stutter? For example `parser.Dir` vs `parser.ParseDir` and `parser.Source` vs `parser.ParseSource`?

Reply:
The weird part is it doesn't really work in my opinion. I don't think anybody would  say `parser source`, but they would say `parse source`. If we wanted to fully remove the stutter, we could rename the package? None of the parsing logic is in here anyway.

I do think that mixing the parsing of literals into this package would  be confusing. As an example, the `go/parser` package is focused only on parsing go files and doesn't have parsing individual portions such as literals. The reason why I placed these functions into `ast` originally is because they were more about constructing an AST rather than directly related to the parser.

Reply:
I think parsing literals should be separate from the AST, as the literal is defined in context of the language not in context of the AST.  Someone can define a different syntax with different literals that creates the same AST.

Reply:
These functions are parsing related. If we want to create separate packages for parsing different kinds of things (files vs literals ) we can but I do not see the value currently. The action of parsing is to construct ast objects from source text. 

As for the stutter I am not worried about it as the method names are clear.

Reply:
If you want to move the parsing functions to the package it's fine. I still have my preference, but it doesn't really matter.

I do wish for these methods to return the raw values instead of the literal though. So `ParseString` should return a `string` and not a `StringLiteral`.